File: src/main/java/com/bot/commands/voice/StopCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;
import datadog.trace.api.Trace;

public class StopCommand extends VoiceCommand {

	public StopCommand() {
		this.name = "stop";
		this.arguments = "";
		this.help = "Stops stream and clears the current playlist";
	}

	@Override
	//@trace(operationName = "executeCommand", resourceName = "Stop")
	protected void executeCommand(CommandEvent commandEvent) {
		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
		if (handler == null) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not connected to a voice channel.");
		}
		else {
			handler.stop();
			commandEvent.reply(commandEvent.getClient().getSuccess() + " Stopped audio stream");
			commandEvent.getGuild().getAudioManager().closeAudioConnection();
		}
	}
}

File: src/main/java/com/bot/commands/voice/StopCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.bot.voice.GuildVoiceProvider
import com.bot.voice.VoiceSendHandler
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class StopCommand : VoiceCommand() {
    val guildVoiceProvider = GuildVoiceProvider.getInstance()
    init {
        name = "stop"
        arguments = ""
        help = "Stops stream and clears the current playlist"
    }

    @Trace(operationName = "executeCommand", resourceName = "Stop")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild.idLong)
        if (voiceConnection == null) {
            commandEvent.reply(commandEvent.client.warning + " I am not connected to a voice channel.")
        } else {
            voiceConnection.cleanupPlayer()
            commandEvent.reply(commandEvent.client.success + " Stopped audio stream")
            commandEvent.guild.audioManager.closeAudioConnection()
        }
    }
}